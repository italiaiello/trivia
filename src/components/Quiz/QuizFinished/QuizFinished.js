import React from 'react'
import Emoji from '../Emoji/Emoji'

const QuizFinished = ({ numQuestions, numCorrectAnswers, resetQuiz }) => {

    const score = Math.round((numCorrectAnswers / numQuestions) * 100)

    return (
        <article>
            <h3>{`You scored ${numCorrectAnswers}/${numQuestions}`}</h3>
            {
                score < 50 ?
                <p>Better luck next time <Emoji symbol="&#128577;" label="happy smiley" /></p>
                :
                (
                    score >= 50 && score < 70 ?
                    <p>Congrats, you passed! <Emoji symbol="&#128522;" label="happy smiley" /></p>
                    :
                    (
                        score >= 70 && score < 85 ?
                        <p>Well done! <Emoji symbol="&#128512;" label="happy smiley" /></p>
                        :
                        <p>Wow! You aced it! <Emoji symbol="&#129299;" label="nerd smiley" /></p>
                    )
                )
            }
            <button className="restartButton" onClick={resetQuiz}>Start Again</button>
        </article>
    )
}

export default QuizFinished