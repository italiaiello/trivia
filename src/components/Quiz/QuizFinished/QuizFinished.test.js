import React from 'react'
import { shallow, render } from 'enzyme'
import QuizFinished from './QuizFinished'

describe("Test.js", () => {

    let wrapper;

    // beforeEach(() => {
    //     wrapper = shallow(<QuizFinished numCorrectAnswers={10} numQuestions={30} />)
    // })

    it('checks if it QuizFinished.js renders properly', () => {
        const wrapper = shallow(<QuizFinished numCorrectAnswers={10} numQuestions={30} />)
        expect(wrapper).toMatchSnapshot()
    })

    it('checks if failed score renders correct text', () => {
        const wrapper = render(<QuizFinished numCorrectAnswers={10} numQuestions={30} />)
        const text = wrapper.find('[id="failedText"]')["0"]["children"][0]["data"].trim()
        expect(text).toEqual("Better luck next time")
    })

})