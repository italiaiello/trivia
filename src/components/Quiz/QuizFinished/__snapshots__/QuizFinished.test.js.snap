// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test.js checks if it QuizFinished.js renders properly 1`] = `ShallowWrapper {}`;
